plugins/optimization-detective/class-od-url-metric.php

```diff
@@ -192,16 +228,28 @@ public function get_timestamp(): float {
 	/**
 	 * Gets elements.
 	 *
-	 * @return ElementData[] Elements.
+	 * @return array Elements.
+	 * @phpstan-return ElementData[]
 	 */
 	public function get_elements(): array {
 		return $this->data['elements'];
 	}
 
+	/**
+	 * Gets web vitals.
+	 *
+	 * @return array Web Vitals.
+	 * @phpstan-return WebVitalData
+	 */
+	public function get_web_vitals(): array {
+		return $this->data['webVitals'];
+	}
```

Comment:
Related: Instead of this, assuming the functionality would live in a standalone plugin, we could provide a simple `get( $key )` method that would allow retrieving the data. Or be "even more clever" and have a `__call()` method that automatically resolves it based on the schema, so that you could call `get_web_vitals()` which would resolve to returning `$this->data['webVitals']`. Maybe the latter is overkill, but one of the two could work.

Reply:
Instead of `__call()` I added `__get()` in #1492. It seems a magic getter is better here. However, for static analysis neither `__call()` nor `__get()` would be helpful, so I added a `get($key)` method as well.